lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -257,148 +238,109 @@ def create_instance(instance_params, api_client):
 
     def powerstate_resume_shutdown():
         instance_power_action(
-            module.params['name'], api_client, "resume")
+            module.params['name'], api_client, Action.RESUMED)
         instance_power_action(
-            module.params['name'], api_client, "shutdown")
+            module.params['name'], api_client, Action.SHUTDOWN)
 
     def powerstate_resume_restart():
         instance_power_action(
-            module.params['name'], api_client, "resume")
+            module.params['name'], api_client, Action.RESUMED)
         instance_power_action(
-            module.params['name'], api_client, "restart")
+            module.params['name'], api_client, Action.RESTARTED)
 
     def powerstate_resume_force_restart():
         instance_power_action(
-            module.params['name'], api_client, "resume")
+            module.params['name'], api_client, Action.RESUMED)
         instance_power_action(
-            module.params['name'], api_client, "force-restart")
+            module.params['name'], api_client, Action.FORCE_RESTARTED)
 
     def powerstate_start_pause():
         instance_power_action(
-            module.params['name'], api_client, "start")
+            module.params['name'], api_client, Action.STARTED)
         instance_power_action(
-            module.params['name'], api_client, "pause")
+            module.params['name'], api_client, Action.PAUSED)
 
     # Current state is first dimension
     # Specified state is second dimension
     power_state_dict = {
-        (STATE_running, ACTION_started): powerstate_do_nothing,
-        (STATE_running, ACTION_shutdown): powerstate_shutdown,
-        (STATE_running, ACTION_stopped): powerstate_stop,
-        (STATE_running, ACTION_restarted): powerstate_restart,
-        (STATE_running, ACTION_force_restarted): powerstate_force_restart,
-        (STATE_running, ACTION_paused): powerstate_pause,
-        (STATE_running, ACTION_absent): powerstate_absent,
-        (STATE_shutdown, ACTION_started): powerstate_start,
-        (STATE_shutdown, ACTION_shutdown): powerstate_do_nothing,
-        (STATE_shutdown, ACTION_stopped): powerstate_do_nothing,
-        (STATE_shutdown, ACTION_restarted): powerstate_start,
-        (STATE_shutdown, ACTION_force_restarted): powerstate_start,
-        (STATE_shutdown, ACTION_paused): powerstate_start_pause,
-        (STATE_shutdown, ACTION_absent): powerstate_absent,
-        (STATE_paused, ACTION_started): powerstate_resume,
-        (STATE_paused, ACTION_shutdown): powerstate_resume_shutdown,
-        (STATE_paused, ACTION_stopped): powerstate_stop,
-        (STATE_paused, ACTION_restarted): powerstate_resume_restart,
-        (STATE_paused, ACTION_force_restarted): powerstate_resume_force_restart,
-        (STATE_paused, ACTION_paused): powerstate_do_nothing,
-        (STATE_paused, ACTION_absent): powerstate_absent
+        (State.RUNNING, Action.STARTED): powerstate_do_nothing,
```

Comment:
You could remove all of these functions. And the values in this mapping could be either the `Action` type or a list of `Action` types which you can iterate over and call `powerstate_start_pause` on.